perf/reader_test.go

```diff
@@ -57,7 +57,7 @@ func TestPerfReader(t *testing.T) {
 		t.Fatal("Can't read samples:", err)
 	}
 
-	want := []byte{1, 2, 3, 4, 4, 0, 0, 0, 0, 0, 0, 0}
+	want := []byte{1, 2, 3, 4, 4, 0, 0, 0}
```

Comment:
If you change `mustOutputSamplesProg(t, 5)` above to `mustOutputSamplesProg(t, 4)` you should see only []byte{1,2,3,4}. Your change will break this test.

Reply:
Nice, great catch. This behaviour is related to your comment above https://github.com/cilium/ebpf/pull/94/files#r424357585.